### DISCUSSION TITLE
Question. what is mean "Remove setNativeProps API" ?

### DISCUSSION THREAD

**Discussion by sacru2red** [necolas/react-native-web#2552] [Q&A]
I saw commit [[change] Remove setNativeProps API](https://github.com/necolas/react-native-web/commit/99287696ec5a5a8f4af6779d700e8dc4942b50d5)

I can't understand this. but I have Issue under

I update rnw 18.12 => 19.6, now

some packages got errors
```js
indicatorRef.current.setNativeProps({....})
```
I wonder It's related or not

Thanks

### Steps to reproduce

react-native-web ^0.19.6
react ^18.2.0
react-native-paper-tabs": "^0.7.0",

**Comment by sacru2red** [comment]
@necolas 
I found you [comment](https://github.com/facebook/react-native/issues/34091#issuecomment-1171762209)

It seems like having message "setNativeProps is deprecated. Please update props using React state instead." on 2023-02-07

I don't think have this issue on now.

1. [#migrating-off-setnativeprops](https://reactnative.dev/docs/new-architecture-library-intro#migrating-off-setnativeprops) part is [gone.](https://github.com/facebook/react-native-website/pull/3489) atfer 0.72 doc
2. RN change-logs has no mention this issue

**Reply by sacru2red** [reply]
so, should we revert [commit](https://github.com/necolas/react-native-web/commit/e68c32770757194af103cca0095c0c204995505b) ?

**Reply by sacru2red** [reply]
I think better way not remove setNativeProps. 
some libs or codes use setNativeProps. If keep delete, this is bring high cost to fix 3rd party lib's code.
I think ["Add setNativeProps to Fabric" commit](https://github.com/facebook/react-native/commit/1d3fa40c59b234f21f516db85c322ec0ed0311e0) is similar with my idea

**Reply by vipulrawat** [reply]
I agree that to provide a better backward compatibility `setNativeProps` should be brought back as many libraries still use the API. As mentioned in the comment above, RN is also bringing it back for the same reason.

**Reply by niklasmachacek** [reply]
We have been using `setNativeProps` to update the text of a `TextInput` inside an animation.

Without `setNativeProps` it is simply not possible to have smooth animations here by updating the React State, as mentioned in [this ticket](https://github.com/facebook/react-native-website/pull/3489).

Please consider reverting the commit and getting `setNativeProps` back in!

**Comment by sacru2red** [comment]
@necolas ping. how about your opinion?

**Comment by nabilfreeman** [comment]
At first look of this thread today I thought this was a terrible change that needed to be reverted.

But, after some further thought and research it looks like setNativeProps is still going to be phased out of React Native.

I think this should eventually be removed, but I'm not sure if removing it ahead of the core team is a good idea. `react-native-web` is a hugely important part of the RN ecosystem and I would expect any changes to be timed with those in RN core for maximum compatibility.

Maybe more of a philosophical & DX question.

Personally I'm going to push through this issue with some patching on the culprit 3rd party library causing this problem, which is quite old anyway.

**Reply by jonathanj** [reply]
https://github.com/facebook/react-native/issues/34391#issuecomment-1783161575

**Comment by niklasmachacek** [comment]
I went ahead and filed a new [issue](https://github.com/necolas/react-native-web/issues/2633) here suggesting that the setNativeProps API should be implemented again, since it is now supported in Fabric after react-native 0.72 as seen [here](https://github.com/facebook/react-native/issues/34391#issuecomment-1783161575).

**Reply by nabilfreeman** [reply]
Yes, if it has been added back to Fabric, then it makes sense to restore it here as well IMO
